Data/Scripts/011_Battle/002_Battler/004_Battler_Statuses.rb

```diff
@@ -260,6 +260,10 @@ def pbInflictStatus(newStatus, newStatusCount = 0, msg = nil, user = nil)
     if abilityActive?
       Battle::AbilityEffects.triggerOnStatusInflicted(self.ability, self, user, newStatus)
     end
+    # Poison Puppeteer
+    if !user.nil? && user.abilityActive?
```

Comment:
`if user&.abilityActive?`

Also, move this ability trigger above the `triggerOnStatusInflicted` one. It makes more sense for user to inflict everything it can before the target uses Synchronize to react to the status problem it got.